### ISSUE TITLE
bot is skipping messages 

### ISSUE THREAD

**Issue by ashimloves** [cellsheet/chlib#40]
I am using this for getting messages from a group.
I am opening the script output window and the group opened in a browser.
I have noticed that it is skipping some messages.
Please ping me if any other info is required to fix this. I am also looking for exact cause.
Thanks.

**Comment by piks** [comment]
From what i understand its most likely the outcome of chatango not sending the second message event (the one with the message id with only like 2 or 3 things in the event line i think its the u event but i might be wrong (or completely reversed)

**Comment by ashimloves** [comment]
Well I have not checked the full code yet. If I find any fix will let you know. Thanks for sharing your inputs though. This is a nice library overall.

**Comment by piks** [comment]
@ashimloves I'm sorry,  I thought I replied to this but, the best way to fix this would be to have it store the messages events (b and u) and wait for its pair to appear or drop if it doesn't within a specified time, that however would cause quite a big change in this lib that I do not personally want to mess with, maybe @Hazerdous would be willing to look into it, I don't see it as a high priority as the messages missed are not too many and the lack of a class to work around with the b event would make me want to pull my hair out.

**Comment by LordYusei** [comment]
From looking into this same issue myself, it's caused by chatango not sending the message event, as Piks already said. Not sure myself on how to fix it without screwing too much with the lib. I don't want to break anything else with the new input. I will let you know if I can figure it out.

**Comment by james-d-elliott** [comment]
I wrote a fix for this over a year ago. It's not overly complicated. There are two ways to tackle it; one being more viable considering the CPU and the other being more viable in quality. I'll make a pull with the quality fix and see what you think, keeping in mind I've not tested it.

**Comment by ashimloves** [comment]
I found something which maybe useful to you. I could have provided code but I am doing it from scratch using Java. Anyways...
Chatango is sending the events separated by "\u0000" and sometimes it's in the same line.
So if you split the whole line by "\u0000" then no event will be missed.
At least this is what I found using Java.
Basically I am using a thread to listen to the socket after login and then the 1st thing I am doing is split the whole message returned by "\u0000" and then processing all of the split messages. Hope this helps.
